compiler/rustc_ast/src/tokenstream.rs

```diff
@@ -107,8 +107,9 @@ where
     CTX: crate::HashStableContext,
 {
     fn hash_stable(&self, hcx: &mut CTX, hasher: &mut StableHasher) {
-        for sub_tt in self.trees() {
-            sub_tt.hash_stable(hcx, hasher);
+        for (tt, spacing) in self.0.iter() {
+            tt.hash_stable(hcx, hasher);
+            spacing.hash_stable(hcx, hasher);
         }
     }
```

Comment:
Question was already raised before: why only hash `TokenTree`s? https://github.com/rust-lang/rust/pull/66279/files#r349855691

Reply:
Maybe this is answer: https://github.com/rust-lang/rust/blob/c0672870491e84362f76ddecd50fa229f9b06dff/compiler/rustc_data_structures/src/stable_hasher.rs#L196-L200